catboost/private/libs/algo/target_classifier.cpp

```diff
@@ -59,7 +61,12 @@ TTargetClassifier BuildTargetClassifier(
         case ELossFunction::RMSE:
         case ELossFunction::MultiRMSE:
             return TTargetClassifier(
-                SelectBorders(target, targetBorderCount, targetBorderType, allowConstLabel),
+                SelectBorders(target, targetBorderCount, targetBorderType, allowConstLabel, targetValuesMayContainNans),
+                targetId);
+        case ELossFunction::MultiRMSEWithMissingValues:
+            targetValuesMayContainNans = true;
+            return TTargetClassifier(
+                SelectBorders(target, targetBorderCount, targetBorderType, allowConstLabel, targetValuesMayContainNans),
```

Comment:
Let's set default value for `targetValuesMayContainNans` in `SelectBorders` to `false` and will pass `true` only here.

Reply:
Fixed.